### ISSUE TITLE
FI_FINALIZER_NULLS_FIELDS false positive

### ISSUE THREAD

**Issue by szegedi** [spotbugs/spotbugs#3207]
I'm getting a `FI_FINALIZER_NULLS_FIELDS` because I'm nulling a field in a method with signature
```
	private void finalize(Gatherer.Downstream<? super List<T>> downstream)
```
However, while this method is named "finalize", it is not an override of `protected void finalize()` in `java.lang.Object`, so this is a false positive.

(For context, I'm implementing a [stream gatherer](https://docs.oracle.com/en/java/javase/23/docs/api/java.base/java/util/stream/Gatherer.html) and had a method named "finalize" for a method handle passed as a "finisher" parameter to `Gatherer.of(initializer, integrator, finisher)`. I'll rename it to "finish" to avoid this false positive, but the point remains that the rule misidentified a method as the class instance finalizer.
